src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php

```diff
@@ -53,47 +68,250 @@ public function __construct(PlatformTopologyRepositoryInterface $platformTopolog
+         */
+        if ($registeredParentInTopology && true === $platformTopology->isLinkedToAnotherServer()) {
+            /**
+             * @var PlatformTopology $platformInformation
+             */
+            $platformInformation = $this->platformTopologyRepository->findPlatformInformation();
+
+            if (null === $platformInformation) {
+                throw new PlatformTopologyException(
+                    _("Platform's mandatory data are missing. Please reinstall your platform.")
+                );
+            }
+            if ('no' === $platformInformation->getIsRemote()) {
                 throw new PlatformTopologyConflictException(
                     sprintf(
-                        _("A Central : '%s'@'%s' is already registered"),
-                        $foundCentralPlatformType->getName(),
-                        $foundCentralPlatformType->getAddress()
+                        _("The platform: '%s'@'%s' is not declared as a 'remote'."),
+                        $platformTopology->getName(),
+                        $platformTopology->getAddress()
+                    )
+                );
+            }
+            if (null === $platformInformation->getAuthorizedMaster()) {
+                throw new PlatformTopologyException(
+                    sprintf(
+                        _("The platform: '%s'@'%s' is not linked to any Central. Please use the wizard first."),
+                        $platformTopology->getName(),
+                        $platformTopology->getAddress()
                     )
                 );
             }
-            $foundCentralInNagiosTable = $this->platformTopologyRepository->findPlatformTopologyNagiosId(
-                $platformTopology->getName()
-            );
-
-            if (null === $foundCentralInNagiosTable) {
-                throw new PlatformTopologyConflictException(
+            if (
+                null === $platformInformation->getApiUsername()
+                || null === $platformInformation->getApiCredentials()
+            ) {
+                throw new PlatformTopologyException(
                     sprintf(
-                        _("The Central type server : '%s'@'%s' does not match the one configured in Centreon"),
+                        _("Central's credentials are missing on: '%s'@'%s'. Please check the Remote Access form."),
                         $platformTopology->getName(),
                         $platformTopology->getAddress()
                     )
                 );
             }
-            $platformTopology->setServerId($foundCentralInNagiosTable->getId());
+
+            /**
+             * Call the API on the n-1 server to register it too
+             */
+            try {
+                // Central's API payloads and URL
+
+                /*
+                 *  TODO check url consistency, protocol and root platform name
```

Comment:
is it still in WIP, or will it be manage in a separate PR ?

Reply:
still wip :'(
specs have been adapted to take care of new cases